cips/cip-26.md

```diff
@@ -0,0 +1,64 @@
+| cip | 26 |
+| - | - |
+| title | Reduce block time to six seconds |
+| description | Proposal to reduce block time on celestia-app to 6 seconds, from 12. |
+| author | Josh Stein ([@jcstein](https://github.com/jcstein)), Rootul Patel ([@rootulp](https://github.com/rootulp))|
+| discussions-to | <https://forum.celestia.org/t/cip-decrease-block-time-to-6-seconds/1836> |
+| status | Draft |
+| type | Standards Track |
+| category | Core |
+| created | 2024-10-09 |
+
+## Abstract
+
+This CIP proposes to reduce the block time on celestia-app to 6 seconds, from 12. This will double the throughput and reduce the time it takes for transactions to be finalized in half. This CIP also proposes to increase the `ttl-num-blocks` parameter in the mempool configuration from 5 to 12 to maintain consistency with the new block time.
+
+## Motivation
+
+The motivation for this CIP stems from a discussion in Core Devs Call 17, where it was proposed to reduce the block time to 6 seconds from 12 seconds.
+
+## Specification
+
+1. The block time in celestia-app SHOULD be reduced from 12 seconds to 6 seconds. Concretely, this implies decreasing `TimeoutCommit` to 4.2 seconds and `TimeoutPropose` to 3.5 seconds.
+    1. The `TimeoutCommit` and `TimeoutPropose` parameters were moved from local config parameters into versioned parameters controlled by the state machine to ensure consistency and correctness across different protocol versions. This change allows the system to adapt dynamically to version-specific requirements, improving overall stability and interoperability during upgrades.
+    1. The change in block time MUST be implemented at a specific block height, which SHALL be determined and agreed upon by the Celestia community via on-chain signaling by validators.
```

Comment:
This sentence reads as if there is a separate signalling mechanism for block time change, however, afaik, signalling is a way to agree on a specific version of celestia-app to be used by validators, and since timeouts rely on the celestia-app version, when version transition happens so does the timeout update (if any). I'd suggest revising this sentence to make this point clear.

Reply:
It would be also beneficial to add a link to the CIP containing the signalling mechanism.

Reply:
signaling mechanism is linked in meta CIP that this is included in, so wondering if we should take out this sentence entirely. wdyt @rootulp ?

Reply:
alternatively if you would like to propose a version of this sentence that makes more sense, would be happy to accept it too